### PR TITLE
Bump jinja version

### PR DESCRIPTION
**Pull request by tino097**
Bump jinja version

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -6,7 +6,7 @@ click==6.7
 fanstatic==0.12
 Flask==0.12.4
 Flask-Babel==0.11.2
-Jinja2==2.8
+Jinja2>=2.8.1
```
Comment (amercader) [inline requirements.in]: Can you pin it to this version? Otherwise people will get 2.10 which might be incompatible

**Comment by adborden** [comment]
@amercader Instead of pinning with `==`, why not use a [compatible version specifier](https://www.python.org/dev/peps/pep-0440/#compatible-release), e.g. `Jinja2~=2.8.1`? This would not allow 2.10, but would pick up patch versions automatically.

**Comment by davidread** [comment]
@adborden Nice suggestion, automatically including patches, although may not be practical in this case.

I did a survey of our deps last summer and could only find one that *declared* it followed semantic versioning. Jinja2 doesn't declare it but seems to do it anyway, as is becoming the convention in python, so it makes it a candidate.

However we had horrible problems with html5lib recently, which had a problem introduced in 0.999999 and fixed in 0.999999999 or something. And the interdependencies get complicated. Like this week pycodestyle 2.5 in python2 needs requests 2.21.0 instead and not 2.20.0, but it's not in its setup.py install_requires because it is different in python3 I believe. The bottom line is I think our requirements.txt should be pinned exactly.

We generate our requirements.txt from requirements.in with pip-compile (from pip-tools), so that sub-dependencies are specified in requirements.txt. So I was hoping that we could use `~=` in requirements.in, but it seems that even the latest version of pip-compile doesn't understand the notation.

So in conclusion:
* `~= `would be nice to use for deps that appear to be following semver, in our requirements.in file
* however we can't use `~=` unless we move on from pip-compile. Happy to hear further suggestions - maybe in another issue.

**Comment by davidread** [comment]
The changelog is clear that there are only 3 little fixes in this update, so pretty safe. http://jinja.pocoo.org/docs/2.10/changelog/#version-2-8-1

I clicked around the UI with this newer version and couldn't see any issues.

So @amercader I'd be happy to see this merged.

**Comment by adborden** [comment]
@davidread thanks for the explanation. In general, I think pinning versions in requirements.txt is problematic for ckan when mixed with extensions. Each extension declares its own requirements.txt with its own pinned dependencies which may conflict with others. `pip -r` has no mechanism to even warn when a conflict happens. This creates an impossible dependency situation. I'm hoping to have more bandwidth to look into these issues in the coming months, so if you have a GH issue documenting various pain points, that would be helpful. I've already opened https://github.com/ckan/ckan/issues/4590 and https://github.com/ckan/ckan/issues/4689 but let me know where the appropriate place to have this discussion is.

@tino097 sorry for hijacking your PR.

**Comment by amercader** [comment]
Let's get this merged to address the Jinja issue and carry on the requirements conversation separately.
